reference/matrix/conv_kernels.cpp

```diff
@@ -24,10 +24,34 @@ namespace conv {
 
 template <typename ValueType>
 void conv(std::shared_ptr<const DefaultExecutor> exec,
-          const matrix::Conv<ValueType>* kernel,
-          const matrix::Dense<ValueType>* b, matrix::Dense<ValueType>* x)
+          const array<ValueType>& kernel, const matrix::Dense<ValueType>* b,
+          matrix::Dense<ValueType>* x)
 {
-    GKO_NOT_IMPLEMENTED;
+    const auto b_size = b->get_size();                 // (N, 1)
+    const auto x_size = x->get_size();                 // (N + K - 1, 1)
+    const auto kernel_size = kernel.get_size();        // K
+    const auto* kernel_ptr = kernel.get_const_data();  // pointer to kernel data
+    int stride = 1;
+    int padding = 2;
+    int output_length = (x_size[0] + 2 * padding - kernel_size) / stride + 1;
+
+    for (gko::size_type i = 0; i < x_size[0]; ++i) {
```

Comment:
because you are using substraction later, so I will say you can use gko::int64 directly as Tobias mentioned

Reply:
also I think start with output will be easier than with input